test/javascripts/lib/raw-event-helper-test.js

```diff
@@ -0,0 +1,35 @@
+import { module, test } from "qunit";
+import { replaceRaw } from "discourse/plugins/discourse-calendar/discourse/lib/raw-event-helper";
+
+module("Unit | Lib | raw-event-helper", function () {
+  test("replaceRaw", function (assert) {
+    const raw = 'Some text [event param1="value1"] more text';
+    const params = {
+      param1: "newValue1",
+      param2: "value2",
+    };
+
+    assert.strictEqual(
+      replaceRaw(params, raw),
+      'Some text [event param1="newValue1" param2="value2"] more text',
+      "it updates existing parameters and adds new ones"
```

Comment:
for consistency I'd remove "it " prefix from assertion descriptions, like in the one below